src/agent0/hyperfuzz/system_fuzz/invariant_checks.py

```diff
@@ -100,21 +113,30 @@ def run_invariant_checks(
                 _check_minimum_share_reserves(pool_state),
                 _check_solvency(pool_state),
                 _check_present_value_greater_than_idle_shares(interface, pool_state),
-                _check_checkpointing_should_never_fail(interface, pool_state),
-                _check_initial_lp_profitable(pool_state),
+                _check_previous_checkpoint_exists(interface, pool_state),
             ]
 
-    for failed, message, data in results:
-        any_check_failed = failed | any_check_failed
+    exception_message_base = ["Continuous Fuzz Bots Invariant Checks"]
+    exception_data_template: dict[str, Any] = {}
+    if crash_report_additional_info is not None:
+        exception_data_template.update(crash_report_additional_info)
+
+    out_exceptions: list[FuzzAssertionException] = []
+    for failed, message, data, log_level in results:
+        if not failed:
+            continue
+        exception_message = exception_message_base.copy()
         if message:
             exception_message.append(message)
+        exception_data = exception_data_template.copy()
         exception_data.update(data)
         exception_data["block_number"] = pool_state.block_number
 
-    # Log additional information if any test failed
-    if any_check_failed:
-        logging.critical("\n".join(exception_message))
+        # Log exception to rollbar
+        assert log_level is not None
```

Comment:
is this for type narrowing or is this possible at run time? if run time make it an `if` --> `raise` conditional.

Reply:
This is type narrowing, log_level should always be set in this case